data_safe_haven/external/api/azure_api.py

```diff
@@ -724,7 +722,7 @@ def remove_blob(
                 f"Removed file [green]{blob_name}[/] from blob storage.",
             )
         except AzureError as exc:
-            msg = f"Blob file [green]'{blob_name}'[/] could not be removed from [green]'{storage_account_name}'[/].\n{exc}"
+            msg = f"Blob file [green]'{blob_name}'[/] could not be removed from [green]'{storage_account_name}'[/]."
```

Comment:
```suggestion
            msg = f"Blob file '{blob_name}' could not be removed from '{storage_account_name}'."
```

as these message now bypass our logger we probably don't want to inject colour formatting.

Reply:
I think there should still be logged.

The `__init__` method of `DataSafeHavenError` calls `logger.error` with the exception message.

Reply:
OK - does this interfere with the traceback output though?

Reply:
I would hope not, as that is also a rich feature. I'll try making a minimal example.

Reply:
Great. If it works let's leave things as they are, if not let's remove the highlighting from exception messages. In either case, please merge once you're happy :).

Reply:
OK, so it looks like the answer is, the markup is processed by the called to log (through the rich handler) but the messages in exceptions and tracebacks are raw strings with the markup syntax.

We could do something like

```
class DataSafeHavenError(Exception):
    def __init__(self, message: str | bytes):
        super().__init__(remove_markup(message))

        # Log exception message as an error
        logger = get_logger()
        logger.error(message)
```

Reply:
I don't think it's a problem to not have coloured output in our exception messages.